lib/dao.js

```diff
@@ -1169,6 +1170,12 @@ DataAccessObject._coerce = function (where) {
                 throw err;
               }
               break;
+            case 'regexp':
+              if (typeof val !== 'string' || val instanceof RegExp) {
```

Comment:
It should be `if (!(typeof val === 'string' || (val instanceof RegExp))) {`.. We complains if the value is not string or RegExp.